plugins/kotlin/jvm-debugger/core/src/org/jetbrains/kotlin/idea/debugger/core/metadataUtil.kt

```diff
@@ -0,0 +1,69 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+
+package org.jetbrains.kotlin.idea.debugger.core
+
+import com.google.gson.Gson
+import com.google.gson.JsonSyntaxException
+import com.intellij.debugger.engine.evaluation.EvaluationContext
+import com.sun.jdi.*
+import kotlinx.metadata.jvm.KotlinClassMetadata
+import org.jetbrains.kotlin.idea.debugger.base.util.wrapEvaluateException
+import org.jetbrains.kotlin.idea.debugger.base.util.wrapIllegalArgumentException
+import kotlin.io.encoding.Base64
+import kotlin.io.encoding.ExperimentalEncodingApi
+
+fun ReferenceType.fetchKotlinMetadata(context: EvaluationContext): KotlinClassMetadata? {
+    val classObject = classObject() ?: return null
```

Comment:
We can cache the result of this function to avoid redundant computations

Reply:
I also thought about creating a metadata cache. We need a cache that would invalidate itself when the debug process stops. The only issue is that I didn't find any 'debug process wise' caches in Intellij IDEA. Only project wide ones.